sycl/source/device_selector.cpp

```diff
@@ -144,7 +143,9 @@ select_device(const DSelectorInvocableType &DeviceSelectorInvocable,
 __SYCL_EXPORT int default_selector_v(const device &dev) {
   // The default selector doesn't reject any devices.
   int Score = 0;
-
+  if (dev.get_backend() == backend::ext_intel_esimd_emulator) {
+    return 0;
```

Comment:
Probably the expected behavior would never pick the esimd emulator in `default_selector` since it doesn't fully support SYCL features.

Reply:
@bso-intel , Any thoughts on Vlad's suggestion?
Should we have the default selector outright reject instead of choosing it only when there are no other devices?
I suppose it is a reasonable assumption to make that there will always be another device in the system for the default emulator to choose from(cpu) so that it will not typically throw a device not found exception?